### PR TITLE
Simplify calling structure

### PR DESCRIPTION
**Pull request by martint**
Returning a Predicate is unnecessary and makes the code harder to follow

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionTreeUtils.java [hunk 1]
```diff
@@ -50,11 +50,11 @@ private ExpressionTreeUtils() {}
...
 
-    private static Predicate<FunctionCall> isAggregationPredicate(Metadata metadata)
+    private static boolean isAggregation(FunctionCall functionCall, Metadata metadata)
     {
-        return ((functionCall) -> (metadata.isAggregationFunction(functionCall.getName())
-                || functionCall.getFilter().isPresent()) && !functionCall.getWindow().isPresent()
-                || functionCall.getOrderBy().isPresent());
+        return ((metadata.isAggregationFunction(functionCall.getName()) || functionCall.getFilter().isPresent())
+                && !functionCall.getWindow().isPresent())
+                || functionCall.getOrderBy().isPresent();
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionTreeUtils.java]: If I am not mistaken, this is equivalent to:
```
boolean isAggregationFunction = metadata.isAggregationFunction(functionCall.getName());
boolean hasFilter = functionCall.getFilter().isPresent();
boolean hasWindow = functionCall.getWindow().isPresent();
boolean hasOrderBy = functionCall.getOrderBy().isPresent();
return ((isAggregationFunction || hasFilter) && !hasWindow) || hasOrderBy;
```

How come non-aggregate function (according to metadata) is reported as an aggregate when filter is defined? I thought one can define filter _on aggregate functions_.

(No change requested, I verified semantics here are not changed.)

File: presto-main/src/main/java/io/prestosql/sql/planner/EqualityInference.java [hunk 2]
```diff
@@ -244,32 +244,30 @@ Expression getScopedCanonical(Expression expression, Predicate<Symbol> symbolSco
...
         }
-        return getCanonical(filter(equalitySets.get(canonicalIndex), symbolToExpressionPredicate(symbolScope)));
+        return getCanonical(filter(equalitySets.get(canonicalIndex), equivalentExpression -> isScoped(equivalentExpression, symbolScope)));
     }
 
-    private static Predicate<Expression> symbolToExpressionPredicate(final Predicate<Symbol> symbolScope)
+    private static boolean isScoped(Expression expression, Predicate<Symbol> symbolScope)
     {
-        return expression -> Iterables.all(SymbolsExtractor.extractUnique(expression), symbolScope);
+        return Iterables.all(SymbolsExtractor.extractUnique(expression), symbolScope);
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/sql/planner/EqualityInference.java]:
```
SymbolsExtractor.extractUnique(expression).stream()
  .allMatch(symbolScope);
```

would be good to remove Guava's `filter` too

**Review by findepi — APPROVED** [review]

File: presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionTreeUtils.java (hunk 1 above)
Comment (martint) [reply presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionTreeUtils.java]: This method needs to be revisited at some point. Currently, it extracts aggregation "candidates" based on whether they are aggregations or how they are used would require they *be* aggregations. I guess it has evolved over time with patches over patches, and that's what we ended up with.

File: presto-main/src/main/java/io/prestosql/sql/planner/EqualityInference.java (hunk 2 above)
Comment (martint) [reply presto-main/src/main/java/io/prestosql/sql/planner/EqualityInference.java]: Good point. I'll do that in a separate change.

**Pull request merged**